File: backend/ArtDuParfum/src/main/java/artduparfum/ArtDuParfum/repository/entity/Address.java
package artduparfum.ArtDuParfum.repository.entity;

import jakarta.persistence.*;
import lombok.*;

@Setter
@Getter
@Table(name = "\"addresses\"")
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Address {

    @Id
    @GeneratedValue(
            strategy = GenerationType.AUTO
    )
    private Long id;

    private String country;
    private String region;
    private String city;
    private String postCode;
    private String address;
    private String phone;

    @OneToOne
    private Order order;
}

File: frontend/lib/components/address.dart
class Address {
  final String country;
  final String region;
  final String city;
  final String postCode;
  final String address;
  final String phone;

  Address({
    required this.country,
    required this.region,
    required this.city,
    required this.postCode,
    required this.address,
    required this.phone,
  });

  Map<String, dynamic> toJson() {
    return {
      "country": country,
      "region": region,
      "city": city,
      "postCode": postCode,
      "address": address,
      "phone": phone,
    };
  }
}
